modules/base/src/processors/volumevoronoisegmentation.cpp

```diff
@@ -0,0 +1,104 @@
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <modules/base/processors/volumevoronoisegmentation.h>
+#include <modules/base/algorithm/volume/volumevoronoi.h>
+
+namespace inviwo {
+
+// The Class Identifier has to be globally unique. Use a reverse DNS naming scheme
+const ProcessorInfo VolumeVoronoiSegmentation::processorInfo_{
+    "org.inviwo.VolumeVoronoiSegmentation",  // Class identifier
+    "Volume Voronoi Segmentation",           // Display name
+    "Undefined",                             // Category
+    CodeState::Experimental,                 // Code state
+    Tags::None,                              // Tags
+};
+const ProcessorInfo VolumeVoronoiSegmentation::getProcessorInfo() const { return processorInfo_; }
+
+VolumeVoronoiSegmentation::VolumeVoronoiSegmentation()
+    : Processor(), volume_("inputVolume"), inputMesh_("inputPositions"), outport_("outport") {
+
+    addPort(volume_);
+    addPort(inputMesh_);
+    addPort(outport_);
+}
+
+void VolumeVoronoiSegmentation::process() {
+    auto inputMesh = inputMesh_.getData();
+
+    // Get seed points
+    // TODO: see if this can be done in a better way
+    auto posIt = util::find_if(inputMesh->getBuffers(), [](const auto& buf) {
+        return buf.first.type == BufferType::PositionAttrib;
+    });
+    if (posIt == inputMesh->getBuffers().end()) {
+        return;
+    }
+    const auto positionRam = posIt->second->getRepresentation<BufferRAM>();
+    std::vector<dvec3> positions;
+    for (std::size_t i = 0; i < positionRam->getSize(); ++i) {
+        positions.push_back(positionRam->getAsDVec3(i));
+    }
+
+    // Get indices for seed points (if exists??)
+    // TODO: should probably have some fallback solution if there are no indices...?
+    auto indexIt = util::find_if(inputMesh->getBuffers(), [](const auto& buf) {
```

Comment:
So the IndexBuffers in the mesh only refer to positions in the position buffer above, that does not really have to have anything to do with atom indices. This is why I think using a `DataFrame` here would make it clearer what indices it is that are being used.

Reply:
Yes, I have changed it to use DataFrame instead!